File: test/SelectionSort.java
package test;

import java.util.Arrays;

public class SelectionSort {
    public static void main(String[] args) {
        int[] array = {10, 40, 20, 30, 70, 60, 40, 30};
        System.out.println(Arrays.toString(sort(array)));
    }

    static int[] sort(int[] array) {

        for (int i = 0; i < array.length - 1; i++) {

            int min = i;
            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[min]) {
                    min = j;
                }
            }

            int temp = array[min];
            array[min] = array[i];
            array[i] = temp;

        }
        return array;
    }
}

File: DSA/hackerrank/SelectionSort.kt
package DSA.hackerrank

import java.util.Arrays
import kotlin.math.max

// Performs well on a small list
fun main() {
    val array = arrayOf(10, 8, 4, 20, 19, 18, 23, 90, 43, 72)
    println(Arrays.toString(selectionSort(array)))
}

fun selectionSort(array: Array<Int>): Array<Int> {

    for (i in array.indices) {
        var maxIndex = 0;

        for (j in 0 until array.size - i) {
            if (array[j] > array[maxIndex]) {
                maxIndex = j
            }
        }

        val temp = array[array.size - 1 - i]
        array[array.size - 1 - i] = array[maxIndex]
        array[maxIndex] = temp
    }

    return array;
}